sbt-app/src/sbt-test/actions/compile-clean/test

```diff
@@ -0,0 +1,22 @@
+$ touch target/cant-touch-this
+
+> Test/products
+$ exists target/out/jvm/scala-2.12.17/compile-clean/classes/A.class
+$ exists target/out/jvm/scala-2.12.17/compile-clean/test-classes/B.class
```

Comment:
I think in sbt 2.x we shouldn't promise there's a `*.class` file. Instead we should promise there's a JAR file with the class `A` inside it.

Reply:
I kind of agree. But that test is checking more than the output of compilation: it also tests `clean`, and `cleanKeepFiles` at the level of individual `.class` file. So I prefer to keep it as it is for now, and to update it later if needed, when we actually change the `products` task.